tests/gzip.rs

```diff
@@ -75,9 +75,9 @@ fn test_gzip_empty_body() {
         .unwrap();
 
     let mut body = ::std::string::String::new();
-    res.read_to_string(&mut body).unwrap();
+    let err = res.read_to_string(&mut body).unwrap_err();
 
-    assert_eq!(body, "");
+    assert_eq!(::std::io::ErrorKind::UnexpectedEof, err.kind());
```

Comment:
This test was specifically trying to prevent this error from happening...

Reply:
I thought it was just to prevent false `UnexpectedEof` error. I can bring `Pending` back.